src/scripts/main.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
+const form = document.querySelector('form');
+
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (element) {
```

Comment:
Checking if the element exists before adding an event listener is a good practice. However, it would be better to throw an error immediately if the element does not exist, instead of wrapping it in a Promise.